website/static/js/wikiTreebeard.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+var Fangorn = require('js/fangorn');
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var m = require('mithril');
+var Treebeard = require('treebeard');
+var $osf = require('js/osfHelpers');
+require('../css/fangorn.css');
+
+function resolveIcon(item) {
+    if (item.children.length > 0) {
+        if (item.open) {
+            return m('i.fa.fa-folder-open-o', ' ');
+        }
+        return m('i.fa.fa-folder-o', ' ');
+    }
+}
+
+function WikiMenu(data) {
+
+    //  Treebeard version
+    var tbOptions = {
+        divID: 'grid',
+        filesData: data,
+        paginate : false,       // Whether the applet starts with pagination or not.
+        paginateToggle : false, // Show the buttons that allow users to switch between scroll and paginate.
+        uploads : false,         // Turns dropzone on/off.
+        hideColumnTitles: true,
+        resolveIcon : resolveIcon,
+        columnTitles: function () {
+            return[{
+                title: 'Name',
+                width: '100%'
+            }];
+        },
+        onload: function() {
+            var tb = this;  // jshint ignore: line
+            for (var i = 0; i < tb.treeData.children.length; i++) {
+                var parent = tb.treeData.children[i];
+                if (parent.data.title === 'Project Wiki Pages') {
+                    tb.updateFolder(null, parent);
+                }
+            }
+        },
+        resolveRows : function (item){
+            var tb = this;
+            var columns = [];
+
+            tb.wiki = {
+                id: window.contextVars.wiki.wikiID,
```

Comment:
Reusable module should not rely on global variables because globals make testing and reusability difficult. Instead, these should be passed in as arguments to the constructor.

The only place globals are permissible is in page modules (those that end in `-page.js`).